tests/unit/views/pacbio/PacbioLibraryBatchCreate.spec.js

```diff
@@ -0,0 +1,356 @@
+import { expect, it, vi } from 'vitest'
+import PacbioLibraryBatchCreate from '@/views/pacbio/PacbioLibraryBatchCreate.vue'
+import { usePacbioLibraryBatchCreateStore } from '@/stores/pacbioLibraryBatchCreate.js'
+
+import PrinterFactory from '@tests/factories/PrinterFactory.js'
+import PacbioTagSetFactory from '@tests/factories/PacbioTagSetFactory.js'
+import PacbioLibraryBatchFactory from '@tests/factories/PacbioLibraryBatchFactory.js'
+import useRootStore from '@/stores'
+
+const pacbioTagSetFactory = PacbioTagSetFactory()
+const printerFactory = PrinterFactory()
+const pacbioLibraryBatchFactory = PacbioLibraryBatchFactory()
+
+const mockShowAlert = vi.fn()
+vi.mock('@/composables/useAlert', () => ({
+  default: () => ({
+    showAlert: mockShowAlert,
+  }),
+}))
+
+function mountWithStore({ state = {}, stubActions = false, plugins = [], props } = {}) {
+  const wrapperObj = mount(PacbioLibraryBatchCreate, {
+    global: {
+      plugins: [
+        createTestingPinia({
+          state,
+          stubActions,
+          plugins,
+        }),
+      ],
+    },
+    props,
+  })
+  const createStoreObj = usePacbioLibraryBatchCreateStore()
+  return { wrapperObj, createStoreObj }
+}
+
+describe('PacbioLibraryBatchCreate.vue', () => {
+  let wrapper, pacbioLibraryBatchCreateStore
+
+  beforeEach(async () => {
+    const plugins = [
+      ({ store }) => {
+        if (store.$id === 'root') {
+          store.api.v2.traction.pacbio.tag_sets.get = vi.fn(
+            () => pacbioTagSetFactory.responses.fetch,
+          )
+          store.api.v2.traction.printers.get = vi.fn(() => printerFactory.responses.fetch)
+        }
+      },
+    ]
+    const { wrapperObj, createStoreObj } = mountWithStore({
+      plugins,
+    })
+    wrapper = wrapperObj
+    pacbioLibraryBatchCreateStore = createStoreObj
+    await flushPromises()
+  })
+
+  describe('building the component', () => {
+    it('displays initial state correctly', async () => {
+      const select = wrapper.find('select[data-type="tag-set-list"]')
+      expect(select.exists()).toBe(true)
+      const options = select.findAll('option')
+      expect(options.length).toEqual(Object.values(pacbioTagSetFactory.storeData.tagSets).length)
+
+      const printerSelect = wrapper.find('select[data-type="printer-options"]')
+      expect(printerSelect.exists()).toBe(true)
+      const printerOptions = printerSelect.findAll('option')
+      expect(printerOptions.length).toEqual(
+        printerFactory.storeData.getPrintersOfType('tube').length,
+      )
+
+      const csvInput = wrapper.find('input[id="csvFileInput"]')
+      expect(csvInput.exists()).toBe(true)
+      const csvPreviewBtn = wrapper.find('button[data-type="csv-preview-btn"]')
+      expect(csvPreviewBtn.exists()).toBe(true)
+      expect(csvPreviewBtn.element.disabled).toBe(true)
+      expect(wrapper.find('div[data-type="csv-preview"]').exists()).toBe(false)
+
+      const resetBtn = wrapper.find('#reset')
```

Comment:
> Advisory: I think generally testing whether a button is there and disabled can lead to flaky tests as you are not testing behaviour. Generally better to test what happens if you press it in which case it might already be covered by e2e tests.
I’ve added a bunch of tests for enabling and disabling checks because this page goes through a lot of state changes depending on different actions. These changes affect the button states, so I thought it as more of a behavior test in which case unit tests are the best fit for this since they’re less flaky than e2e tests. Also, then we don’t need to cover all of it in e2e tests, which keeps things consistent with how other pages are set up—unit tests for detailed behaviors and e2e tests for the bigger picture. Hope it makes sense.

Reply:
Ok. It might be worth adding this as a composable so we can dynamically define the state and state changes and then have centralise the tests for it. Maybe worth a technical debt story?